WordPress/Classes/Models/BasePost.m

```diff
@@ -181,7 +182,7 @@ - (void)setDateCreated:(NSDate *)localDate
      If the date is now or in the past, and the status is scheduled, set the status
      to published.
      */
-    if (self.date_created_gmt == nil) {
+    if (self.date_created_gmt == nil || [self.date_created_gmt isEqualToDate:self.dateModified]) {
```

Comment:
> Maybe we should consider adding dateModified to AbstractPost in this PR, considering it's not used in ReaderPost? (to avoid changing the model twice)

I think dateModified belongs in AbstractPost, for sure.  We need to move it as part of #5523. But as long as we're having to reference the property from BasePost it needs to belong to BasePost.  I can't bring myself have parent class calling code in one of its descendants. :/